cukinia

```diff
@@ -363,17 +363,30 @@ _cukinia_process()
 {
 	local pname="$1"
 	local puser="$2"
-	local ret
-
+	local puser_uid
+	
 	_cukinia_prepare "Checking process \"$pname\" as ${puser:-any user}"
-	ret=$(ps -a -o user,comm |
-			awk '($2 == "'$pname'") && ($1 ~ /^'${puser:-.*}'$/) {
-			    print;
-			}')
-	[ -n "$ret" ] && return 0 || return 1
+	
+	if [ -n "$puser" ]; then
+		if ! puser_uid="$(id -u "$puser" 2>/dev/null)" ; then 
+			return 1
+		fi
+	fi
+	
+	# loop in /proc to get process name and optionally process owner
+	for status in /proc/[0-9]*/status; do
+		# check by process name
+		[ "$(grep Name "$status" | cut -f 2)" = "$pname" ] || continue
```

Comment:
This is okay.

We can improve a bit by removing the `| cut -f2` using directly what it is expected in the `grep` expression.

``` bash
		[ "$(grep -q "Name:\t$pname" "$status" ] || continue
```
or `[[:space:]]` if it is available in busybox.
``` bash
		[ "$(grep -q "Name:[[:space:]]$pname" "$status" ] || continue
```

Reply:
I would keep it like it is if you don't mind.

Reply:
Okay.

The only drawback I see is that we do not fully rely on grep and it's return code.

By the way my snippet was wrong again. It was supposed to be as easy as:

``` bash
grep -q "Name:\t$pname" "$status" || continue
```